### PR TITLE
Phaser renderer remove pixi region

### PR DESCRIPTION
**Pull request by sloont**

### CODE REVIEW

File: src/index.ejs [hunk 1]
```diff
@@ -157,10 +157,10 @@
         </script>
 
         <!-- below loads all uncompiled game files (your source code changes will reflect) -->
-        <script type="text/javascript" src="/engine/loader.js"></script>
+        <!-- <script type="text/javascript" src="/engine/loader.js"></script> -->
 
         <!-- below loads the compiled game file (much faster, your source code changes won't reflect until you compile again)-->
```
Comment (sloont) [inline src/index.ejs:162]: TODO: revert this

File: src/gameClasses/Region.js [hunk 2]
```diff
@@ -25,7 +25,7 @@ var Region = IgeEntityPhysics.extend({
 			}
 
 			this._stats.currentBody = {
-				type: 'static',
+				type: 'dynamic',
```
Comment (sloont) [inline src/gameClasses/Region.js:28]: TODO: Revert

File: ts/src/renderer/phaser/classes/PhaserRegion.ts [hunk 3]
```diff
@@ -9,12 +9,23 @@ class PhaserRegion extends Phaser.GameObjects.Graphics {
...
+		// const stats = this.region._stats;
 
 		// draw rectangle
 		const width = this.width = stats.width;
 		const height = this.height = stats.height;
-		this.fillStyle(0xFF0000, 0.4);
+		// Phaser wants a number for these
+		this.fillStyle(
+			Number(`0x${stats.inside.substring(1)}`) || 0xffffff,
+			stats.alpha / 100 || 0.4
```
Comment (sloont) [inline ts/src/renderer/phaser/classes/PhaserRegion.ts:27]: default value for `stats.alpha` should stay,  default for `stats.inside` should be removed. We should only emit `create-region` if there is a value for `.inside` on the `Region` entity.

File: ts/types/EntityStats.d.ts [hunk 4]
```diff
@@ -18,7 +18,10 @@ declare interface EntityStats {
 		x: number,
 		y: number,
 		width: number,
-		height: number
+		height: number,
+		inside?: string,
+		outside?: string,
```
Comment (sloont) [inline ts/types/EntityStats.d.ts:23]: remove `outside`

File: ts/types/Region.d.ts [hunk 5]
```diff
@@ -1,3 +1,3 @@
 declare class Region extends IgeEntityPhysics {
-    regionUi: RegionUi;
+	_id: string;
```
Comment (sloont) [inline ts/types/Region.d.ts:2]: only being used for a console.log

**Comment by sloont** [comment]
`npm run debug --game=6283aeccfc669705284994ae`

**Comment by pavle-goloskokovic** [comment]
can you merge `phaser-renderer` into this branch, I'm seeing projectile changes in the diffs

File: ts/src/renderer/phaser/classes/PhaserRegion.ts [hunk 6]
```diff
@@ -1,62 +1,64 @@
...
-		const stats = this.region._stats.default
+		const stats = this.region._stats.default;
 
-		// draw rectangle
-		const width = this.width = stats.width;
-		const height = this.height = stats.height;
-		this.fillStyle(0xFF0000, 0.4);
+		// Phaser wants a number for these
+		this.fillStyle(
+			Number(`0x${stats.inside.substring(1)}`),
```
Comment (pavle-goloskokovic) [inline ts/src/renderer/phaser/classes/PhaserRegion.ts:16]: nice

File: ts/src/renderer/phaser/classes/PhaserRegion.ts [hunk 7]
```diff
@@ -1,62 +1,64 @@
 class PhaserRegion extends Phaser.GameObjects.Graphics {
 
-	private width: number;
-	private height: number;
+	private updateDimensionsListener: EvtListener;
```
Comment (pavle-goloskokovic) [inline ts/src/renderer/phaser/classes/PhaserRegion.ts:3]: could this be `sizeListener`, a bit less verbose

File: ts/src/renderer/phaser/classes/PhaserRegion.ts [hunk 8]
```diff
@@ -1,62 +1,64 @@
...
 
 		scene.add.existing(this);
 		scene.events.on('update', this.update, this);
-	}
-
-	update (/*time: number, delta: number*/): void {
 
-		const region = this.region;
-		const container = region.regionUi._pixiContainer;
+		this.updateDimensionsListener = region.on('update-region-dimensions', () => {
```
Comment (pavle-goloskokovic) [inline ts/src/renderer/phaser/classes/PhaserRegion.ts:32]: also, can event name be simply `size` ? it is emitted on region, so we already know what is getting updated

File: src/gameClasses/Region.js [hunk 9]
```diff
@@ -55,59 +52,41 @@ var Region = IgeEntityPhysics.extend({
...
+				// IgeEntity.streamMode(val)
+				// 1 is 'automatic' streaming
 				self.streamMode(1);
 			} else if (ige.isClient) {
-				if (typeof mode === 'string' && mode === 'sandbox') {
-					delete self._stats.value;
+				if ((mode === 'play' && self._stats.default.inside) || mode === 'sandbox') {
+					// o.O TODO: Remove /refactor
+					ige.entitiesToRender.trackEntityById[entityIdFromServer] = this;
+					ige.client.emit('create-region', this);
```
Comment (sloont) [inline src/gameClasses/Region.js:68]: does this emit from self now instead of client?

File: ts/src/renderer/phaser/classes/PhaserRegion.ts [hunk 10]
```diff
@@ -1,62 +1,64 @@
...
-			height
+			stats.width,
+			stats.height
 		);
 
 		this.x = stats.x;
 		this.y = stats.y;
 
 		scene.add.existing(this);
 		scene.events.on('update', this.update, this);
```
Comment (pavle-goloskokovic) [inline ts/src/renderer/phaser/classes/PhaserRegion.ts:30]: `update` event listener should be removed

File: ts/src/renderer/phaser/classes/PhaserRegion.ts [hunk 11]
```diff
@@ -1,62 +1,64 @@
...
+		this.updateDimensionsListener = region.on('update-region-dimensions', () => {
 
-		if (region._destroyed || container._destroyed) {
-			this.scene.events.off('update', this.update, this);
-			this.destroy();
-			return;
-		}
-
-		const stats = this.region._stats.default
+			const stats = this.region._stats.default;
```
Comment (pavle-goloskokovic) [inline ts/src/renderer/phaser/classes/PhaserRegion.ts:34]: lines 12-27 and 34-51 are basically duplicates. can you move this logic into a class method and then invoke it once in constructor and on each size update event?

**Review by pavle-goloskokovic — CHANGES_REQUESTED** [review]

File: ts/src/renderer/phaser/classes/PhaserRegion.ts (hunk 11 above)
Comment (sloont) [reply ts/src/renderer/phaser/classes/PhaserRegion.ts:34]: heard and agree 👍

File: ts/src/renderer/phaser/classes/PhaserRegion.ts (hunk 8 above)
Comment (sloont) [reply ts/src/renderer/phaser/classes/PhaserRegion.ts:32]: thing is its also sending positional updates

Comment (pavle-goloskokovic) [reply ts/src/renderer/phaser/classes/PhaserRegion.ts:32]: then maybe use `transform` instead of `size`?

Comment (sloont) [reply ts/src/renderer/phaser/classes/PhaserRegion.ts:32]: thats actually exactly what I did :D

**Review by pavle-goloskokovic — APPROVED** [review]

**Review by Grenagar — APPROVED** [review]

**Pull request merged**